curriculum/challenges/english/25-front-end-development/lab-recipe-page/668f08ea07b99b1f4a91acab.md

```diff
@@ -147,7 +147,27 @@ assert.exists(document.querySelector('img'));
 All your `img` elements should have a valid `src` attribute and value.
 
 ```js
-assert.isAbove(document.querySelector('img')?.src.length, 0);
+const img = document.querySelector('img');
+const rawSrc = img?.getAttribute('src');
+const resolvedSrc = img?.src;
+const re = new RegExp(window.location.href, "ig")
+
+assert.isAbove(rawSrc?.trim().length, 0, "The 'src' attribute must be explicitly set.");
+assert(resolvedSrc != window.location.href, "The 'src' should not be the same as the current page URL.");
```

Comment:
```suggestion

```
this is redundant, it would also fail the check below, but the check below is broader, so let's keep that